### DISCUSSION TITLE
ruby object_id vs tables with columns named object_id

### DISCUSSION THREAD

**Discussion by pgorgita** [jeremyevans/sequel#2252] [General]
Hi Jeremy,

It is a tricky one, I guess.

I've created models for already existing Oracle and Postgres DBs that sadly use columns named 'object_id' as id.

Both with models and with datasets I've encountered the following Issue that you might reproduce with:

```ruby
DB.create_table(:test) { Integer :object_id }
DB[:test].insert({object_id: 123456789})
DB[:test].all               => [{:object_id=>123456789}]
DB[:test].first.object_id   => 8160
DB[:test].first[:object_id] => 123456789
```

As workaround I've been using ```def_column_alias :oid, :object_id``` but there is a catch: 
  Used in associations `oid` will work at an instance level, but not at Class level using eager_graph for instance,.

As a solution I've defined 2 associations for each reltion, one to use with instance associations using `oid`, and another to deal with eager_graph for the same exact association using the column name `object_id`.

I doesn't seem a bug to me.

Have you ever had such Issue, or have you a suggestion to better address such Issue?

Thanks in advance,

**Comment by jeremyevans** [comment]
If you can change the schema, that's definitely a easiest way to work around the issue, but I'm guessing you don't have the option.  You probably want to use the column_conflicts plugin.

**Reply by pgorgita** [reply]
Man, you're both fast and omnipresent!

Thanks for the great work shared here, and for your help.

I'll try it, and check if the 'performance hit' does not affect my use cases.

Cheers,

Paulo

**Comment by pgorgita** [comment]
As expected, it works.

Thanks
